src/sql/workbench/common/actions.ts

```diff
@@ -302,6 +302,12 @@ export class BackupAction extends Task {
 	}
 
 	runTask(accessor: ServicesAccessor, profile: IConnectionProfile): TPromise<void> {
+		if (!profile) {
+			let objectExplorerService = accessor.get<IObjectExplorerService>(IObjectExplorerService);
+			let connectionManagementService = accessor.get<IConnectionManagementService>(IConnectionManagementService);
+			let workbenchEditorService = accessor.get<IEditorService>(IEditorService);
+			profile = TaskUtilities.getCurrentGlobalConnection(objectExplorerService, connectionManagementService, workbenchEditorService);
+		}
```

Comment:
Just seeing this now, but I think its still possible for the profile to be undefined. Probably want to add another check and just fail quietly in that scenario. Otherwise this will throw errors.